From c228010cd81988478834e51b1025e4cb0cd7ac72 Mon Sep 17 00:00:00 2001
From: Detlev Casanova <detlev.casanova@collabora.com>
Date: Fri, 15 Nov 2024 11:20:40 -0500
Subject: [PATCH 04/90] [FROM-ML] dt-bindings: display: vop2: Add VP clock
 resets

Add the documentation for VOP2 video ports reset clocks.
One reset can be set per video port.

Reviewed-by: Conor Dooley <conor.dooley@microchip.com>
Signed-off-by: Detlev Casanova <detlev.casanova@collabora.com>
Signed-off-by: Cristian Ciocaltea <cristian.ciocaltea@collabora.com>
---
 .../display/rockchip/rockchip-vop2.yaml       | 48 +++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/Documentation/devicetree/bindings/display/rockchip/rockchip-vop2.yaml b/Documentation/devicetree/bindings/display/rockchip/rockchip-vop2.yaml
index 93da1fb9ad..8a8993fea8 100644
--- a/Documentation/devicetree/bindings/display/rockchip/rockchip-vop2.yaml
+++ b/Documentation/devicetree/bindings/display/rockchip/rockchip-vop2.yaml
@@ -82,6 +82,26 @@ properties:
       - {}
       - {}
 
+  resets:
+    minItems: 5
+    items:
+      - description: AXI clock reset.
+      - description: AHB clock reset.
+      - description: Pixel clock reset for video port 0.
+      - description: Pixel clock reset for video port 1.
+      - description: Pixel clock reset for video port 2.
+      - description: Pixel clock reset for video port 3.
+
+  reset-names:
+    minItems: 5
+    items:
+      - const: aclk
+      - const: hclk
+      - const: dclk_vp0
+      - const: dclk_vp1
+      - const: dclk_vp2
+      - const: dclk_vp3
+
   rockchip,grf:
     $ref: /schemas/types.yaml#/definitions/phandle
     description:
@@ -153,6 +173,12 @@ allOf:
 
         interrupt-names: false
 
+        resets:
+          maxItems: 5
+
+        reset-names:
+          maxItems: 5
+
         ports:
           required:
             - port@0
@@ -200,6 +226,12 @@ allOf:
         interrupt-names:
           minItems: 4
 
+        resets:
+          maxItems: 5
+
+        reset-names:
+          maxItems: 5
+
         ports:
           required:
             - port@0
@@ -251,6 +283,12 @@ allOf:
 
         interrupt-names: false
 
+        resets:
+          minItems: 6
+
+        reset-names:
+          minItems: 6
+
         ports:
           required:
             - port@0
@@ -289,6 +327,16 @@ examples:
                               "dclk_vp0",
                               "dclk_vp1",
                               "dclk_vp2";
+                resets = <&cru SRST_A_VOP>,
+                         <&cru SRST_H_VOP>,
+                         <&cru SRST_VOP0>,
+                         <&cru SRST_VOP1>,
+                         <&cru SRST_VOP2>;
+                reset-names = "aclk",
+                              "hclk",
+                              "dclk_vp0",
+                              "dclk_vp1",
+                              "dclk_vp2";
                 power-domains = <&power RK3568_PD_VO>;
                 rockchip,grf = <&grf>;
                 iommus = <&vop_mmu>;
-- 
2.46.0

